File: 프로그래머스/Summer Winter Coding(~2018)/스킬트리.js
// 스킬트리
function solution(skill, skill_trees) {
    let answer = 0;
    for(let tree of skill_trees){
        let queue = skill.split("");
        let check = true;
        tree.split("").map((e) => {
            if(queue.includes(e)){
                if(queue[0] == e) queue.shift();
                else check = false;
            }
        })
        if(check) answer += 1;
    }
    return answer;
}

File: 프로그래머스/Summer Winter Coding(~2018)/스킬트리.py
# 스킬 트리
from collections import deque
def solution(skill, skill_trees):
    answer = 0
    for tree in skill_trees:
        check = True
        queue = deque(list(skill))
        for t in tree:
            if t in queue and queue[0] == t:
                queue.popleft()
            if t in queue and queue[0] != t:
                check = False
                break
        if check:
            answer += 1
    
    return answer